apps/wallet-dashboard/components/Dialogs/migration/views/ConfirmMigrationView.tsx

```diff
@@ -0,0 +1,156 @@
+// Copyright (c) 2024 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+import React from 'react';
+import { VirtualList } from '@/components';
+import { useCurrentAccount } from '@iota/dapp-kit';
+import { IotaObjectData } from '@iota/iota-sdk/client';
+import {
+    Button,
+    Header,
+    InfoBox,
+    InfoBoxStyle,
+    InfoBoxType,
+    KeyValueInfo,
+    LoadingIndicator,
+    Panel,
+    Title,
+    TitleSize,
+} from '@iota/apps-ui-kit';
+import { useGroupedMigrationObjectsByExpirationDate } from '@/hooks';
+import { Loader, Warning } from '@iota/ui-icons';
+import { Collapsible, useFormatCoin } from '@iota/core';
+import { IOTA_TYPE_ARG } from '@iota/iota-sdk/utils';
+import { summarizeMigratableObjectValues } from '@/lib/utils';
+import { MigrationObjectDetailsCard } from '@/components/migration/migration-object-details-card';
+import { DialogLayout, DialogLayoutBody, DialogLayoutFooter } from '../../layout';
+import { Transaction } from '@iota/iota-sdk/transactions';
+
+interface ConfirmMigrationViewProps {
+    basicOutputObjects: IotaObjectData[] | undefined;
+    nftOutputObjects: IotaObjectData[] | undefined;
+    onSuccess: () => void;
+    setOpen: (bool: boolean) => void;
+    isTimelocked: boolean;
+    migrateData:
+        | {
+              transaction: Transaction;
+              gasBudget: string | number | null;
+          }
+        | undefined;
+    isMigrationPending: boolean;
+    isMigrationError: boolean;
+    isSendingTransaction: boolean;
+}
+
+export function ConfirmMigrationView({
+    basicOutputObjects = [],
+    nftOutputObjects = [],
+    onSuccess,
+    setOpen,
+    isTimelocked,
+    migrateData,
+    isMigrationPending,
+    isMigrationError,
+    isSendingTransaction,
+}: ConfirmMigrationViewProps): JSX.Element {
+    const account = useCurrentAccount();
+
+    const {
+        data: resolvedObjects = [],
+        isLoading,
+        error: isGroupedMigrationError,
+    } = useGroupedMigrationObjectsByExpirationDate(
+        [...basicOutputObjects, ...nftOutputObjects],
+        isTimelocked,
+    );
+
+    const { totalNotOwnedStorageDepositReturnAmount } = summarizeMigratableObjectValues({
```

Comment:
something weird is happening here, the IOTA to be migrated do not correspond to those signed in the tx. Maybe this is a bit more logical, we can look at it in another task
![image](https://github.com/user-attachments/assets/50b78f13-e88c-4572-83b4-af39701b6aea)

Reply:
If resource repo is used to mint the NFT outputs than they also hold the balance of 1 Iota each so that is where the extra iota balance is probably coming from

Reply:
we are taking those IOTA into account, look [here](https://github.com/iotaledger/iota/blob/develop/apps/wallet-dashboard/lib/utils/migration/groupStardustObjectsByMigrationStatus.ts#L78)

Reply:
https://github.com/iotaledger/iota/issues/4562